test/integration/job/job_test.go

```diff
@@ -1040,6 +1040,132 @@ func TestIndexedJob(t *testing.T) {
+			defer featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, features.JobPodFailurePolicy, tc.usePodFailurePolicy)()
+
+			closeFn, restConfig, clientSet, ns := setup(t, "pod-replacement-policy")
+			defer closeFn()
+			ctx, cancel := startJobControllerAndWaitForCaches(restConfig)
+			defer cancel()
+			resetMetrics()
+
+			// Set up initial Job in Indexed completion mode.
+			mode := batchv1.IndexedCompletion
+			jobSpec := batchv1.JobSpec{
+				Parallelism:    pointer.Int32Ptr(initialCompletions),
+				Completions:    pointer.Int32Ptr(initialCompletions),
+				CompletionMode: &mode,
+			}
+			if tc.podReplacementPolicy != "" {
+				jobSpec.PodReplacementPolicy = (*batchv1.PodReplacementPolicy)(&tc.podReplacementPolicy)
+			}
+			if tc.usePodFailurePolicy {
+				podFailurePolicy := &batchv1.PodFailurePolicy{
+					Rules: []batchv1.PodFailurePolicyRule{
+						{
+							Action: batchv1.PodFailurePolicyActionFailJob,
+							OnExitCodes: &batchv1.PodFailurePolicyOnExitCodesRequirement{
+								Operator: batchv1.PodFailurePolicyOnExitCodesOpIn,
+								Values:   []int32{5},
+							},
+						},
+					},
+				}
+				jobSpec.PodFailurePolicy = podFailurePolicy
+			}
+			jobObj, err := createJobWithDefaults(ctx, clientSet, ns.Name, &batchv1.Job{
+				Spec: jobSpec,
+			})
+			if err != nil {
+				t.Fatalf("Failed to create Job: %v", err)
+			}
+			jobClient := clientSet.BatchV1().Jobs(jobObj.Namespace)
+
+			// Wait for pods to start up.
+			err = wait.PollImmediate(5*time.Millisecond, wait.ForeverTestTimeout, func() (done bool, err error) {
+				job, err := jobClient.Get(ctx, jobObj.Name, metav1.GetOptions{})
+				if err != nil {
+					return false, err
+				}
+				if job.Status.Active == int32(initialCompletions) {
+					return true, nil
+				}
+				return false, nil
+			})
+			if err != nil {
+				t.Fatalf("Error waiting for Job pods to become active: %v", err)
+			}
+			pods, errList := clientSet.CoreV1().Pods(ns.Namespace).List(ctx, metav1.ListOptions{})
+			if errList != nil {
+				t.Fatalf("Failed to list pods: %v", errList)
+			}
+			updatePod(t, clientSet, pods.Items, func(pod *v1.Pod) {
+				pod.Finalizers = append(pod.Finalizers, "fake.example.com/blockDeletion")
+			})
+			err = clientSet.CoreV1().Pods(ns.Name).DeleteCollection(ctx,
+				metav1.DeleteOptions{},
+				metav1.ListOptions{
+					Limit: 1000,
+				})
+			if err != nil {
+				t.Fatalf("Failed to cleanup Pods: %v", err)
+			}
+
+			podsDelete, errList2 := clientSet.CoreV1().Pods(ns.Namespace).List(ctx, metav1.ListOptions{})
+			if errList != nil {
+				t.Fatalf("Failed to list pods: %v", errList2)
+			}
+			for _, val := range podsDelete.Items {
+				if val.DeletionTimestamp == nil {
+					t.Fatalf("Deletion not registered.")
+				}
+			}
+
+			validateJobsPodsStatusOnly(ctx, t, clientSet, jobObj, podsByStatus{
+				Terminating: pointer.Int32Ptr(tc.wantTerminating),
+				Failed:      int(tc.wantTerminating),
```

Comment:
It isn't clear to me why Failed=terminating. I prefer to have `wantFailed`.

Reply:
Added `wantFailed` as a paramter